### PR TITLE
Rebase Crypten

### PR DESCRIPTION
**Pull request by gmuraru**

### CODE REVIEW

File: syft/workers/virtual.py [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
+        """send message to worker location"""
+        if self.message_pending_time > 0:
+            if self.verbose:
+                print(f"pending time of {self.message_pending_time} seconds to send message...")
+            sleep(self.message_pending_time)
+
         return location._recv_msg(message)
 
     def _recv_msg(self, message: bin) -> bin:
+        """receieve message"""
```
Comment (LaRiffle) [inline syft/workers/virtual.py]: Typo

File: test/workers/test_virtual.py [hunk 2]
```diff
@@ -25,6 +25,9 @@ def test_send_msg():
     # get pointer to local worker
     me = sy.torch.hook.local_worker
 
+    # pending time to simulate lantency (optional)
+    me.message_pending_time = 0.1
```
Comment (LaRiffle) [inline test/workers/test_virtual.py]: As me is global, shouldn’t we set it back to zero at the end of the test?

File: test/workers/test_virtual.py [hunk 3]
```diff
@@ -34,10 +37,14 @@ def test_send_msg():
...
 
     # Send data to bob
+    start_time = time()
     me.send_msg(ObjectMessage(obj), bob)
+    elapsed_time = time() - start_time
 
     # ensure that object is now on bob's machine
     assert obj_id in bob._objects
+    # ensure that object was sent 0.1 secs later
+    assert abs(elapsed_time - me.message_pending_time) < 0.1
```
Comment (LaRiffle) [inline test/workers/test_virtual.py]: That’s confusing why not just ensure rlapsed_time > 0.1 ?

File: test/workers/test_virtual.py (hunk 2 above)
Comment (gmuraru) [reply test/workers/test_virtual.py]: It should be set

File: test/workers/test_virtual.py (hunk 3 above)
Comment (gmuraru) [reply test/workers/test_virtual.py]: You are right, will do that.

File: syft/workers/virtual.py (hunk 1 above)
Comment (gmuraru) [reply syft/workers/virtual.py]: Done

File: test/workers/test_virtual.py (hunk 3 above)
Comment (gmuraru) [reply test/workers/test_virtual.py]: Done

**Pull request merged**